pubsub/aws/snssqs/topics_locker.go

```diff
@@ -0,0 +1,48 @@
+package snssqs
+
+import (
+	"sync"
+
+	"github.com/puzpuzpuz/xsync/v2"
+)
+
+var (
+	lockManager     *TopicsLockManager
+	lockManagerOnce sync.Once
+)
+
+// TopicsLockManager is a singleton for fine-grained locking, to prevent the component r/w operations
+// from locking the entire component out when performing operations on different topics.
+type TopicsLockManager struct {
+	xLockMap *xsync.MapOf[string, *sync.Mutex]
```

Comment:
Why using xsync rather than just a sync.Map here?

Reply:
Also note v2 of the package is deprecated: https://github.com/puzpuzpuz/xsync

Reply:
> Why using xsync rather than just a sync.Map here?

xsync supports the `LoadOrCompute` and `Compute` methods which are crucial here

> Also note v2 of the package is deprecated: https://github.com/puzpuzpuz/xsync

v3.0.0 was released 3 days ago I don't know if it is as stable as the semver suggests

Reply:
> v3.0.0 was released 3 days ago I don't know if it is as stable as the semver suggests

We have some time. SInce v2 is unmaintained now, let's use the new version please!